IdentityCore/src/shared_device/MSIDDarwinNotificationListener.h

```diff
@@ -0,0 +1,42 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef void (^MSIDDarwinNotificationCallback)(NSError * _Nullable error);
+
+@interface MSIDDarwinNotificationListener : NSObject
```

Comment:
Should this have a more specific name (something like `MSIDGlobalSignoutDarwinNotificationListener`)? 
I mean this is specific about waiting for the account changed notification so that the global signout can happen, but the name suggests it is generic and you could eventually subscribe to another notification.

Reply:
Might want to change the name of the notification as well in that case?

Reply:
Yeah, initially I figured that for extensibility reasons, the Darwin notification listener class might someday create listeners other than `createSharedDeviceAccountChangeListener`. I don't know if that will be the case though...